### DISCUSSION TITLE
Better Piracy Protection

### DISCUSSION THREAD

**Discussion by sarsamurmu** [zixpo/candybar#210] [Ideas]
So, I was thinking about creating a new system for icon packs that would improve protection against piracy.

**This is just a concept for now. Things could change later.**

## Concept
The system would be like how KWGT + KWGT Packs work. There will be an icon pack manager app, and other icon packs will have their own app, these icon packs can't be used with a launcher directly. Unlike traditional icon packs, these icon packs will not include the icons and XML files. These icon packs would provide an API key, the icon pack manager will make sure that the icon pack is genuine and then it will generate an APK after fetching the data from the server using the API key. The user will install the APK to use the icon pack.

### Can't hackers upload the generated APK for everyone to use?
The generated APK will only contain the icons for the user's apps. So, even they share the APK, many icons will be missing for other users.
### So how can users pick alternative icons from the icon pack?
This can be done with the icon pack manager
### What is the downside?
Some users may find it hard to use.

## Notes
- The icon pack manager app will be closed source. Only the icon pack template will be open source.
- Since there is a server involved, the service will be paid. Pricing will be $2/month/app.
- Single icon packs can contain more than one pack.
- As files are being downloaded from the web, there is no size limit. Icon packs can exceed 200MB.
- This will be a large project, so I think it will take 1 year to complete.

What do you all think? Would you use this?

If you have any questions you can ask on this thread or on telegram. This project would not work without your support.

Subscribe (check the left sidebar) to receive updates about this project.

**Comment by Donnnno** [comment]
Interesting idea for people with premium icon packs! 
But in reality, when you publish through Google Play, you won't be able to do this, because of restrictions on the `REQUEST_INSTALL_PACKAGES` permission.

I don't see Google handing out the consent to small icon pack devs to allow them using this 'high-risk permissions'.
Even with ten million+ downloads, [you can still get rejected](https://www.reddit.com/r/iconPackStudio/comments/107bh3m/ips_22_changelog_and_download_link/).

https://support.google.com/googleplay/android-developer/answer/12085295

**Reply by sarsamurmu** [reply]
Yeah, even though this idea was good, it is not possible to do this in reality because of the restrictions.

**Reply by Donnnno** [reply]
It's really unfortunate, this solution would have been great. However, I get that Google doesn't want this permission to be used by everyone security wise. 
The alternative would be that people would have to install the icon pack via a third-party store, but that also means that the user base would be small.

**Comment by moertel** [comment]
Everyone who wants to hack the system will eventually accomplish it. In ways we never thought possible. 😅 I'd say let's concentrate on bringing value to the genuine users who care about our work, rather than wasting effort on those who clearly don't.

Edit: Let me specify: If there's a way for the two apps to communicate, there will be a way for a third-party to intercept this communication. Especially if they root their phone, you'll be 100% unable to keep any API keys secret. Imagine what a lucrative target the server would become. Scary!
